src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
This is bad idea, but I know, it's current implementation is also weird.

Will be replaced with #4581 I hope.